test/e2e/util.go

```diff
@@ -1337,3 +1338,69 @@ func matchIPv4StringFamily(ipStrings []string) (string, error) {
 func matchIPv6StringFamily(ipStrings []string) (string, error) {
 	return util.MatchIPStringFamily(true /*ipv6*/, ipStrings)
 }
+
+// This is a replacement for e2epod.DeletePodWithWait(), which does not handle pods that
+// may be automatically restarted (https://issues.k8s.io/126785)
+func deletePodWithWait(ctx context.Context, c clientset.Interface, pod *v1.Pod) error {
+	if pod == nil {
+		return nil
+	}
+	if pod.UID == "" {
+		// We only recurse into deletePodWithWaitByName when UID is *not* set, to
+		// avoid infinite loops.
+		return deletePodWithWaitByName(ctx, c, pod.Name, pod.Namespace)
+	}
+	oldUID := pod.UID
```

Comment:
why is this oldUID? its the current pod.UID of the podObj we are about to delete right?

Reply:
hm... yeah, this code looks like I originally wrote something else, and then rewrote it, and there are still bits of the old code leaking through...